### PR TITLE
Add gas limit to voice message execution and ACK on out-of-gas.

### PR DESCRIPTION
**Pull request by 0xekez**
This adds a manually tuned gas limit to voice packet handling so that even if the packet causes an out-of-gas error, the module reserves enough gas to return an ACK informing the caller of the out-of-gas error.

### CODE REVIEW

**Review by JakeHartnell — APPROVED** [review]
Nice!

**Comment by 0xekez** [comment]
This is blocked on two things:

1. At least in the IBC testing framework, gas usage is non-deterministic. See https://github.com/DA0-DA0/polytone/issues/23 which describes how I was able to write a test to reproduce this.
2. In response to gas usage being non-dererministic, I want to test this change against a real chain because I am paranoid, but at the moment interchaintest doesn't use the gas limits you specify so I can't instantiate the voice module: https://github.com/strangelove-ventures/interchaintest/issues/482

**Comment by 0xekez** [comment]
@Art3miX - lmk what you think of this. i reorganized the tests, set up CI, found a bug, and made some progress reproducing the handshake bug in interchaintest so i'm now pretty sure that's a problem with IBC and not with our handshake process.

**Comment by 0xekez** [comment]
now blocked on more SDK quirks:tm: https://github.com/cosmos/ibc-go/issues/3428

File: tests/strangelove/incompatible_handshake_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package strangelove
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// Tests that a note may only every connect to a voice, and a voice
```
Comment (Art3miX) [inline tests/strangelove/incompatible_handshake_test.go:9]:
```suggestion
// Tests that a note may only ever connect to a voice, and a voice
```

**Review by Art3miX — COMMENTED** [review]
Awesome as always, few small nits

File: justfile [hunk 2]
```diff
@@ -4,9 +4,6 @@ build:
 test:
     cargo test
 
-lint:
-    cargo clippy --all-targets -- -D warnings
-
```
[the remark this answers is not in the document]
Comment (0xekez) [reply justfile:9]: i'm having some trouble running it on my local machine, though i think this is a problem with my rust install and not the command. will add back.

File: tests/simtests/functionality_test.go [hunk 3]
```diff
@@ -190,3 +190,80 @@ func TestHandshakeBetweenSameModule(t *testing.T) {
...
+}
+
+// Tests executing a message on the remote chain, checking the
+// callback, and then executing another message.
+//
+// This tests that we correctly save proxies and reuse them upon
+// another message being executed. Before this test, we were not
+// saving proxies after instantiating them which would cause
+// `codespace wasm: 15` (duplicate instantiation) errors upon
+// attempting to execute a second message with the same account.g
```
[the remark this answers is not in the document]
Comment (0xekez) [reply tests/simtests/functionality_test.go:229]: i know 😮 shows how hard writing tests is!

File: tests/simtests/functionality_test.go [hunk 4]
```diff
@@ -190,3 +190,80 @@ func TestHandshakeBetweenSameModule(t *testing.T) {
...
+			},
+		},
+	}
+
+	callback, err := suite.RoundtripExecute(t, path, &accountA, []any{gasCosmosgMsg})
+
+	require.NoError(t, err, "out-of-gas should not error")
+	require.Equal(t, Callback{
+		Error: "codespace: sdk, code: 11",
+	}, callback, "out-of-gas should return an ACK")
```
[the remark this answers is not in the document]
Comment (0xekez) [reply tests/simtests/functionality_test.go:219]: thanks for pointing this out. added a comment!

**Pull request merged**